core/src/main/java/org/neo4j/ogm/context/EntityRowModelMapper.java

```diff
@@ -32,14 +36,17 @@
 
         RowModel model;
         while ((model = response.next()) != null) {
-
             if (model.variables().length > 1) {
                 throw new RuntimeException(
                     "Scalar response queries must only return one column. Make sure your cypher query only returns one item.");
             }
             for (int i = 0; i < model.variables().length; i++) {
-                Object o = model.getValues()[0];
-                result.add((T) Utils.coerceTypes(type, o));
+                final Object o = model.getValues()[0];
```

Comment:
Extract this block into a method. The loop does two things: convert the value if possible and adding the value to the list. These are two levels of abstraction.